src/components/modals/TextInputModal.tsx

```diff
@@ -93,7 +93,7 @@ export function TextInputModal(props: Props) {
   return (
     <ThemedModal warning={warning} bridge={bridge} onCancel={() => bridge.resolve(undefined)}>
       {title != null ? <ModalTitle>{title}</ModalTitle> : null}
-      {message != null ? <ModalMessage>{message}</ModalMessage> : null}
+      {message != null ? typeof message === 'string' ? <ModalMessage>{message}</ModalMessage> : <>{message}</> : null}
```

Comment:
This can be simplified to `{typeof message === 'string' ? <ModalMessage>{message}</ModalMessage> : message}`. The `null` will pass through, and the fragment isn't needed in this context. Fragments are mainly just needed to make `return` statements happy.